src/docfx/lib/path/FilePath.cs

```diff
@@ -50,46 +52,47 @@ public FilePath(string path)
             _hashCode = HashCode.Combine(Path, DependencyName, Origin, IsGitCommit);
         }
 
-        private FilePath(FileOrigin origin, PathString path, PathString dependencyName, bool isGitCommit)
+        private FilePath(FileOrigin origin, PathString path, PathString dependencyName, bool isGitCommit, MonikerList monikers)
         {
             Path = path;
             Origin = origin;
             DependencyName = dependencyName;
             IsGitCommit = isGitCommit;
             Format = GetFormat(path);
+            Monikers = monikers;
 
-            _hashCode = HashCode.Combine(Path, DependencyName, Origin, IsGitCommit);
+            _hashCode = HashCode.Combine(Path, DependencyName, Origin, IsGitCommit, Monikers);
         }
 
         public static FilePath Content(PathString path)
         {
             Debug.Assert(!System.IO.Path.IsPathRooted(path));
-            return new FilePath(FileOrigin.Main, path, default, default);
+            return new FilePath(FileOrigin.Main, path, default, default, default);
         }
 
-        public static FilePath Redirection(PathString path)
+        public static FilePath Redirection(PathString path, MonikerList monikers = default)
```

Comment:
nit: consider remove the optional constructor parameter to ensure redirection has monikers
```suggestion
        public static FilePath Redirection(PathString path, MonikerList monikers)
```